src/main/java/com/google/cloud/genomics/dataflow/readers/bam/Reader.java

```diff
@@ -97,12 +113,18 @@ void openFile() throws IOException {
     }
   }
 
+  static boolean isWrongSequence(SAMRecord record, Filter filter, 
+      String referenceName) {
+    return 
+        (filter == Filter.UNMAPPED_ONLY && !record.getReadUnmappedFlag()) ||
+        (filter == Filter.MAPPED_ONLY && record.getReadUnmappedFlag()) || 
+        (filter != Filter.UNMAPPED_ONLY && 
```

Comment:
Sorry @iliat, where is the added comment?  I think my eyes are playing tricks on me this morning, or I just didn't have enough coffee.

thx,
~p